src/main/java/org/folio/service/file/SplitFileProcessingService.java

```diff
@@ -39,16 +61,168 @@ public class SplitFileProcessingService {
 
   private static final Logger LOGGER = LogManager.getLogger();
 
+  private Vertx vertx;
+
+  private FileSplitService fileSplitService;
+  private MinioStorageService minioStorageService;
+
   private DataImportQueueItemDao queueItemDao;
   private UploadDefinitionService uploadDefinitionService;
 
+  private ParallelFileChunkingProcessor fileProcessor;
+
   @Autowired
   public SplitFileProcessingService(
+    Vertx vertx,
+    FileSplitService fileSplitService,
+    MinioStorageService minioStorageService,
     DataImportQueueItemDao queueItemDao,
-    UploadDefinitionService uploadDefinitionService
+    UploadDefinitionService uploadDefinitionService,
+    ParallelFileChunkingProcessor fileProcessor
   ) {
+    this.vertx = vertx;
+
+    this.fileSplitService = fileSplitService;
+    this.minioStorageService = minioStorageService;
+
     this.queueItemDao = queueItemDao;
     this.uploadDefinitionService = uploadDefinitionService;
+
+    this.fileProcessor = fileProcessor;
+  }
+
+  /** Start a job based on information passed to the /processFiles endpoint */
+  public Future<Void> startJob(
+    ProcessFilesRqDto entity,
+    ChangeManagerClient client,
+    OkapiConnectionParams params
+  ) {
+    return initializeJob(entity, client)
+      .compose(splitPieces ->
+        CompositeFuture.all(
+          splitPieces
+            .entrySet()
+            .stream()
+            .map(entry ->
+              initializeChildren(entity, client, params, entry.getValue())
+            )
+            .collect(Collectors.toList())
+        )
+      )
+      // do this after everything has been queued successfully
+      .compose(v ->
+        uploadDefinitionService.updateBlocking(
+          entity.getUploadDefinition().getId(),
+          definition ->
+            Future.succeededFuture(
+              definition.withStatus(UploadDefinition.Status.COMPLETED)
+            ),
+          params.getTenantId()
+        )
+      )
+      .andThen(v -> LOGGER.info("Job split and queued successfully!"))
```

Comment:
would be helpful to use onSuccess() method. With andThen() method the logger will be invoked even if the current future fails and as a result, the success message will appear along with the error message in the log in case of failures, which in turn may confuse during logs investigation.

Reply:
ah whoops, I thought `andThen` was a success-only handler. will fix in a separate PR